src/lib/key_store.c

```diff
@@ -33,62 +33,277 @@
 #include "rnp.h"
 #include "rnpdefs.h"
 #include "key_store.h"
+#include "key_store_internal.h"
 #include "key_store_pgp.h"
+#include "key_store_kbx.h"
 #include "key_store_ssh.h"
 #include "packet-print.h"
 #include "packet-key.h"
 #include "packet.h"
 
-#include <regex.h>
+#include <sys/types.h>
+#include <sys/param.h>
+
 #include <stdio.h>
-#include <stdlib.h>
+#include <regex.h>
 #include <string.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/* read any gpg config file */
+static int
+conffile(rnp_t *rnp, char *homedir, char *userid, size_t length)
+{
+    regmatch_t matchv[10];
+    regex_t    keyre;
+    char       buf[BUFSIZ];
+    FILE *     fp;
+
+    __PGP_USED(rnp);
+    (void) snprintf(buf, sizeof(buf), "%s/gpg.conf", homedir);
+    if ((fp = fopen(buf, "r")) == NULL) {
+        return 0;
+    }
+    (void) memset(&keyre, 0x0, sizeof(keyre));
+    (void) regcomp(&keyre, "^[ \t]*default-key[ \t]+([0-9a-zA-F]+)", REG_EXTENDED);
+    while (fgets(buf, (int) sizeof(buf), fp) != NULL) {
+        if (regexec(&keyre, buf, 10, matchv, 0) == 0) {
+            (void) memcpy(userid,
+                          &buf[(int) matchv[1].rm_so],
+                          MIN((unsigned) (matchv[1].rm_eo - matchv[1].rm_so), length));
```

Comment:
this part of code is moved from [here](https://github.com/riboseinc/rnp/pull/223/files/8b53a319fd46e60fb58cb89c650afc71b665f124#diff-4a64c94981e2d23ec7dd253c41ddf5dbL118).